runners/core-java/src/main/java/org/apache/beam/runners/core/SimpleDoFnRunner.java

```diff
@@ -95,11 +108,20 @@ public SimpleDoFnRunner(
       AggregatorFactory aggregatorFactory,
       WindowingStrategy<?, ?> windowingStrategy) {
     this.fn = fn;
-    this.observesWindow =
-        DoFnSignatures.getSignature(fn.getClass()).processElement().observesWindow();
+    this.signature = DoFnSignatures.getSignature(fn.getClass());
```

Comment:
Yes, I think so. Probably in what we have been talking about, `DoFnRunner` can be altered to accept a `DoFnInvoker` and a `DoFnSignature`. We are not ready for this yet, because of `OldDoFn` needs, I think.